src/view/placeholder.js

```diff
@@ -153,6 +153,12 @@ export function needsPlaceholder( element ) {
 		return false;
 	}
 
+	// If the element is a Widget, always consider it a non-empty and thus not needing a placeholder.
+	// https://github.com/ckeditor/ckeditor5/issues/1684
+	if ( element.getCustomProperty( 'widget' ) ) {
```

Comment:
On the second thought, that would be the only place in the codebase other than `isWidget()` from `ckeditor5-widget` that calls `getCustomProperty( 'widget' )` 😛

Maybe we could use `ckeditor5-widget` as an engine dependency after all. I'm not sure about consequences, though. cc @Reinmar.

Reply:
TBH I have nothing against the current state (`element.getCustomProperty( 'widget' )`) and some improvements in i26 as long as we don't change dependencies in i25.